src/lib/cache/cacheManager.ts

```diff
@@ -0,0 +1,356 @@
+    const conn = getConn();
+    const inMemoryCache = getInMemoryCache();
+    if (!conn || !inMemoryCache) return;
+
+    // Step 1: Get all commands from the database
+    const prefixCommands = await PrefixCommand.find();
+    // Step 2: Get all commands from the cache
+    const cacheKeys = await inMemoryCache.store.keys();
+    // Step 3: Loop over cached commands
+    for (const key of cacheKeys) {
+        if (key.startsWith(`${memoryCachePrefixCommand}:`)) {
+            const checkCommand = key.split(':')[1];
+            // Step 3.a: Check if cached command exists in the database list
+            let found = false;
+            for (const dbCommand of prefixCommands) {
+                const { name: dbCommandName, aliases: dbCommandAliases } = dbCommand;
+                if (dbCommandName.toLowerCase() === checkCommand.toLowerCase() || dbCommandAliases.includes(checkCommand)) {
+                    found = true;
+                    break;
+                }
+            }
+            // Step 3.b: If not found, remove from cache
+            if (!found) {
+                Logger.debug(`Removing command or alias ${checkCommand} from cache`);
+                // eslint-disable-next-line no-await-in-loop
+                await inMemoryCache.del(key);
+            }
+        }
+    }
+    // Step 4: Loop over database commands and update cache
+    for (const dbCommand of prefixCommands) {
+        // eslint-disable-next-line no-await-in-loop
+        await loadSinglePrefixCommandToCache(dbCommand);
+    }
+}
+
+/**
+ * Prefix Command Version Cache Management Functions
+ */
+
+export async function clearSinglePrefixCommandVersionCache(version: IPrefixCommandVersion) {
+    const inMemoryCache = getInMemoryCache();
+    if (!inMemoryCache) return;
+
+    const { alias, _id: versionId } = version;
+    Logger.debug(`Clearing cache for command version alias "${alias}"`);
+    await inMemoryCache.del(`${memoryCachePrefixVersion}:${alias.toLowerCase()}`);
+    await inMemoryCache.del(`${memoryCachePrefixVersion}:${versionId}`);
+}
+
+export async function loadSinglePrefixCommandVersionToCache(version: IPrefixCommandVersion) {
+    const inMemoryCache = getInMemoryCache();
+    if (!inMemoryCache) return;
+
+    const { alias, _id: versionId } = version;
+    Logger.debug(`Loading version with alias ${alias} to cache`);
+    await inMemoryCache.set(`${memoryCachePrefixVersion}:${alias.toLowerCase()}`, version.toObject());
+    await inMemoryCache.set(`${memoryCachePrefixVersion}:${versionId}`, version.toObject());
+}
+
+export async function loadAllPrefixCommandVersionsToCache() {
+    const conn = getConn();
+    const inMemoryCache = getInMemoryCache();
+    if (!conn || !inMemoryCache) return;
+
+    const PrefixCommandVersions = await PrefixCommandVersion.find();
+    for (const version of PrefixCommandVersions) {
+        // eslint-disable-next-line no-await-in-loop
+        await loadSinglePrefixCommandVersionToCache(version);
+    }
```

Comment:
```suggestion
    await Promise.all(PrefixCommandVersions.map((version) => loadSinglePrefixCommandVersionToCache(version)));
```
I'm running out of ideas of what to write here 😂